auth-table-based-service/src/main/java/io/stargate/auth/table/AuthTableBasedService.java

```diff
@@ -119,29 +145,50 @@ public String createToken(String key, String secret) throws UnauthorizedExceptio
       logger.error("Failed to add new token", e);
       throw new RuntimeException(e);
     }
+  }
 
-    return token.toString();
+  private String queryUsername(String key) throws ExecutionException, InterruptedException {
+    ResultSet resultSet =
+        dataStore
+            .query()
+            .select()
+            .column("role")
+            .from("system_auth", "roles")
+            .where("role", WhereCondition.Predicate.Eq, key)
+            .execute();
+
+    if (resultSet.isEmpty()) {
+      throw new RuntimeException(String.format("Provided username %s is incorrect", key));
+    }
+
+    Row row = resultSet.one();
+    if (!row.has("role")) {
+      throw new RuntimeException(String.format("Provided username %s is incorrect", key));
+    }
+
+    return row.getString("role");
   }
 
-  private String queryHashedPassword(String key) throws ExecutionException, InterruptedException {
+  private String queryHashedPassword(String hashedPassword)
+      throws ExecutionException, InterruptedException {
     ResultSet resultSet =
         dataStore
             .query()
             .select()
             .column("salted_hash")
             .from("system_auth", "roles")
-            .where("role", WhereCondition.Predicate.Eq, key)
+            .where("role", WhereCondition.Predicate.Eq, hashedPassword)
             .execute();
 
     if (resultSet.isEmpty()) {
       throw new RuntimeException(
-          String.format("Provided username %s and/or password are incorrect", key));
+          String.format("Provided username %s and/or password are incorrect", hashedPassword));
```

Comment:
Is it intentional to expose the hashedPassword in these error messages?

Reply:
That variable was actually misnamed, it's the username. I'll fix that.